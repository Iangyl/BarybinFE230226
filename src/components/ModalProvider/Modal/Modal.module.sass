@import 'styles/styles.sass'

.modal
  background-color: #fff
  padding: 8px 10px 20px 10px
  border: 1px solid #cecece

.closeContainer
  display: flex
  justify-content: end
  .closeBtn
    &:hover
      cursor: pointer

@include screen-size(375px)
  .modal
    width: 300px
