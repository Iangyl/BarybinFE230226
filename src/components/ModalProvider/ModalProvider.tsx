import React, { useContext, createContext, useState } from 'react';
import { IModalContext } from './ModalProvider.types';
import ModalSpace from './ModalSpace/ModalSpace';

const ModalContext = createContext<IModalContext | null>(null);

const error = <div>Something went wrong!</div>;

const ModalProvider = ({
  children,
}: {
  children: JSX.Element | JSX.Element[];
}) => {
  const [isOpen, setIsOpen] = useState(false);
  const [element, setElement] = useState<JSX.Element | JSX.Element[]>(error);

  const closeModalWindow = () => setIsOpen(false);
  const invokeModalWindow = (element: JSX.Element | JSX.Element[]) => {
    setElement(element);
    setIsOpen(true);
  };

  return (
    <ModalContext.Provider value={{ closeModalWindow, invokeModalWindow }}>
      {isOpen && <ModalSpace element={element} />}
      {children}
    </ModalContext.Provider>
  );
};

export const useModal = () => useContext(ModalContext)!;

export default ModalProvider;
