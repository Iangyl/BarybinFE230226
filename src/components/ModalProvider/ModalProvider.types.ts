export interface IModalContext {
  invokeModalWindow: (element: JSX.Element | JSX.Element[]) => void;
  closeModalWindow: () => void;
}
