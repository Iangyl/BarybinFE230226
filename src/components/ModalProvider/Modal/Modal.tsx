import React from 'react';
import Close from 'assets/Close';
import { useModal } from '../ModalProvider';

import styles from './Modal.module.sass';

const CloseContainer = ({ onClick }: { onClick: () => void }) => (
  <div className={styles.closeContainer}>
    <button className={styles.closeBtn} onClick={onClick}>
      <Close width={24} height={24} />
    </button>
  </div>
);

const Modal = ({
  children,
}: {
  children: JSX.Element | JSX.Element[] | string;
}) => {
  const { closeModalWindow } = useModal();

  return (
    <div className={styles.modal}>
      <CloseContainer onClick={closeModalWindow} />
      {children}
    </div>
  );
};

export default Modal;
