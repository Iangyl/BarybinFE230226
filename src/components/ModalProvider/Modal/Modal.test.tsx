import React from 'react';
import Modal from './Modal';
import { useModal } from '../ModalProvider';
import { render } from '@testing-library/react';
import { IModalContext } from '../ModalProvider.types';

jest.mock('../ModalProvider');
const mockedHook = useModal as jest.Mock<IModalContext>;

test('renders a modal correctly', () => {
  mockedHook.mockReturnValue({
    closeModalWindow: () => console.log('close'),
    invokeModalWindow: () => console.log('open'),
  });
  const { container } = render(<Modal>View</Modal>);
  expect(container.firstChild).toMatchSnapshot();
});
