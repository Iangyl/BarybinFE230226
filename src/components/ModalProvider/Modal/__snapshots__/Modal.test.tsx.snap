// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a modal correctly 1`] = `
<div
  class="modal"
>
  <div
    class="closeContainer"
  >
    <button
      class="closeBtn"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12 7 7m5 5 5 5m-5-5 5-5m-5 5-5 5"
          stroke="#000000"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
  View
</div>
`;
